smart-frontend/app/src/api/adh/clusterInfo.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { PaginateCollection } from '@models/collection';
+import { httpClient } from '@api/httpClient';
+import type { AdhClusterNodesFilter, AdhClusterNode } from '@models/adh';
+import type { PaginationParams, SortParams } from '@models/table';
+import { prepareDateRange, prepareQueryParams } from '@utils/requestUtils';
+import qs from 'qs';
+
+export class AdhClusterInfoApi {
+  public static async getNodes(
+    { registrationTime, ...filter }: AdhClusterNodesFilter,
+    sortParams?: SortParams,
+    paginationParams?: PaginationParams,
+  ): Promise<PaginateCollection<AdhClusterNode>> {
+    const { from: registrationTimeFrom, to: registrationTimeTo } = prepareDateRange(registrationTime);
+
+    const queryParams = prepareQueryParams(
+      {
+        ...filter,
+        registrationTime: { registrationTimeFrom, registrationTimeTo },
```

Comment:
there is case when filter can be empty ( when  wegit hosts list for filter on actions page. Should catch it, and not send  registrationTime in this case

Reply:
if `registrationTimeFrom, registrationTimeTo` are undefined then `registrationTime` will not be sent in query